{
    "source": "BDSA",
    "name": "BDSA-2019-5305",
    "title": "Python typed_ast and cpython Vulnerable to Denial-of-Service (DoS) via Out-of-Bounds Read in 'ast_for_arguments'",
    "description": "An out-of-bounds read was discovered in typed_ast and its implementation within cypthon. An attacker who is able to cause a python interpreter to parse Python source could exploit this in order to cause a denial-of-service (DoS) condition.",
    "technicalDescription": "This issue occurs in the `ast_for_arguments()` function within the `Python/ast.c` file. It was mitigated by the introduction of more robust checks designed to prevent an out-of-bounds read occurring.",
    "publishedDate": "2022-03-04T14:29:31.394Z",
    "updatedDate": "2022-03-04T14:29:31.369Z",
    "disclosureDate": "2019-03-31T00:00:00.000Z",
    "solution": "Fixed in typed_ast [**1.3.2**](https://github.com/python/typed_ast/releases/tag/1.3.2) by [this](https://github.com/python/typed_ast/commit/dc317ac9cff859aa84eeabe03fb5004982545b3b) commit.\n\nThe latest stable releases of typed_ast can be found [here](https://github.com/python/typed_ast/tags).\n\nFixed in cpython [**3.8.0a4**](https://github.com/python/cpython/releases/tag/v3.8.0a4) by [this](https://github.com/python/cpython/commit/a4d78362397fc3bced6ea80fbc7b5f4827aec55e) commit.\n\nThe latest stable releases of cpython can be found [here](https://github.com/python/cpython/tags).",
    "severity": "MEDIUM",
    "cvss2": {
        "baseScore": 5.0,
        "impactSubscore": 2.9,
        "exploitabilitySubscore": 10.0,
        "severity": "LOW",
        "accessVector": "NETWORK",
        "accessComplexity": "LOW",
        "authentication": "NONE",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "PARTIAL",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 3.7
        },
        "vector": "(AV:N/AC:L/Au:N/C:N/I:N/A:P/E:U/RL:OF/RC:C)"
    },
    "cvss3": {
        "baseScore": 7.5,
        "impactSubscore": 3.6,
        "exploitabilitySubscore": 3.9,
        "severity": "MEDIUM",
        "attackVector": "NETWORK",
        "attackComplexity": "LOW",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "HIGH",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "NONE",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 6.5
        },
        "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H/E:U/RL:O/RC:C"
    },
    "useCvss3": true,
    "classifications": [],
    "zeroDay": false,
    "underReview": false,
    "parentAdvisory": false,
    "workaround": "",
    "vendorFixDate": "2019-04-01T00:00:00.000Z",
    "bdsaTags": [],
    "overallScore": 6.5,
    "_meta": {
        "allow": [
            "GET"
        ],
        "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2019-5305",
        "links": [
            {
                "rel": "cwes",
                "href": "https://trendmicro2.app.blackduck.com/api/cwes/CWE-125"
            },
            {
                "rel": "related-vulnerability",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/CVE-2019-19275",
                "label": "NVD"
            },
            {
                "rel": "bdsa-ranges",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2019-5305/ranges"
            },
            {
                "rel": "reference",
                "href": "https://bugs.python.org/issue36495",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/commit/a4d78362397fc3bced6ea80fbc7b5f4827aec55e",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/commit/dcfcd146f8e6fc5c2fc16a4c192a0c5f5ca8c53c",
                "label": "INTRODUCTORY_COMMIT"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/pull/12641",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/releases/tag/v3.8.0a4",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/tags",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/typed_ast/commit/156afcb26c198e162504a57caddfe0acd9ed7dce",
                "label": "INTRODUCTORY_COMMIT"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/typed_ast/commit/dc317ac9cff859aa84eeabe03fb5004982545b3b",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/typed_ast/releases/tag/1.3.2",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/typed_ast/tags",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "default-remediation-status",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2019-5305/default-remediation-status"
            }
        ]
    }
}